src/mongocxx/test/client_side_encryption.cpp

```diff
@@ -681,8 +689,292 @@ TEST_CASE("Views are prohibited", "[client_side_encryption]") {
+    }
+
+    // Using client_encrypted, find the inserted document from db.coll to a variable
+    // named corpus_decrypted. Since it should have been automatically decrypted, assert
+    // the document exactly matches corpus.
+    auto res = encrypted_coll.find_one({});
+    REQUIRE(res);
+    auto corpus_decrypted = res.value();
+    REQUIRE(corpus_decrypted == corpus);
+
+    // Load corpus/corpus_encrypted.json to a variable named corpus_encrypted_expected.
+    // Using client find the inserted document from db.coll to a variable named
+    // corpus_encrypted_actual.
+    res = client["db"]["coll"].find_one({});
+    REQUIRE(res);
+    auto corpus_encrypted_actual = res.value();
+
+    // Iterate over each field of corpus_encrypted_expected and check the following:
+    for (bsoncxx::document::element ele : corpus_encrypted_expected.view()) {
+        auto field_name_view = ele.key();
+        std::string field_name{field_name_view};
+
+        std::vector<std::string> copied_fields = {"_id", "altname_aws", "altname_local"};
+        if (std::find(copied_fields.begin(), copied_fields.end(), field_name) !=
+            copied_fields.end()) {
+            continue;
+        }
+
+        auto subdoc_val = ele.get_document();
+        auto subdoc = subdoc_val.view();
+
+        auto algo = subdoc["algo"].get_utf8().value;
+        auto allowed = subdoc["allowed"].get_bool().value;
+        auto value = subdoc["value"].get_value();
+
+        auto actual_field = corpus_encrypted_actual.view()[field_name];
+
+        // If the algo is det, that the value equals the value of the corresponding field
+        // in corpus_encrypted_actual.
+        if (algo == stdx::string_view{"det"}) {
+            REQUIRE(value == actual_field["value"].get_value());
+        }
+
+        // If the algo is rand and allowed is true, that the value does not equal the value
+        // of the corresponding field in corpus_encrypted_actual.
+        if (algo == stdx::string_view{"rand"} && allowed) {
+            REQUIRE(value != actual_field["value"].get_value());
+        }
+
+        // If allowed is true, decrypt the value with client_encryption. Decrypt the value of
+        // the corresponding field of corpus_encrypted and validate that they are both equal.
+        if (allowed) {
+            auto decrypted_actual = client_encryption.decrypt(actual_field["value"].get_value());
+            auto decrypted_expected = client_encryption.decrypt(value);
+            REQUIRE(decrypted_expected == decrypted_actual);
+        } else {
+            // If allowed is false, validate the value exactly equals the value of the corresponding
+            // field of corpus (neither was encrypted).
+            REQUIRE(value == corpus.view()[field_name]["value"].get_value());
+        }
+    }
+}
+
 TEST_CASE("Corpus", "[client_side_encryption]") {
-    // TODO CXX-2022
+    if (!mongocxx::test_util::should_run_client_side_encryption_test()) {
```

Comment:
Looks like this needs the server version skip, too, to run successfully against 4.0 servers on evergreen